server.py

```diff
@@ -25,39 +25,44 @@ def add_datum(self, datum):
                                                          time.time())
 
     def get_datum(self):
-        datum = {}
-        for i in range(len(self._calculators)):
-            datum.update({f"patient-{i}": self._calculators[i].descriptors})
-        return datum
+        return {f"patient-{i}": self._calculators[i].descriptors
+                for i in range(len(self._calculators))}
 
 
 class Communicator():
 
     def __init__(self, port=5000):
-        self._socket = zmq.Context().socket(zmq.PUB)
+        self._context = zmq.Context()
+        self._socket = self._context.socket(zmq.PUB)
         self._socket.bind(f"tcp://*:{port}")
 
+    def close(self):
+        self._socket.close()
+        self._context.term()
+
     def publish_message(self, message):
         self._socket.send_multipart([b"",
                                      json.dumps(message).encode("ascii")])
 
 
 def main():
     sensors = Sensors()
-    sensor_data = sensors.poll()
     calculator = Calculator()
     communicator = Communicator()
-    running = True
-    while running:
-        try:
+    try:
+        while True:
```

Comment:
Why did `running` go away?

Reply:
well there was never an instance where `running` could be False, and I'm intending on my next PR to be something that fixes this.  I suppose I'll revert till the next PR.